rfc/rfc-0000-supply-chain-tracing.md

```diff
@@ -0,0 +1,182 @@
+      lastTransitionTime: "2022-02-16T03:29:52Z"
+    - name: revision
+      value: main/3d42c19a618bb8fc13f72178b8b5e214a2f989c4
+      lastTransitionTime: "2022-02-16T03:29:52Z"
+    observedGeneration: 1
+  - name: image-builder
+    stampedRef:
+      apiVersion: kpack.io/v1alpha2
+      kind: Image
+      namespace: default
+      name: my-workload
+    templateRef:
+      apiVersion: carto.run/v1
+      kind: ClusterImageTemplate
+      name: image
+    inputs:
+    - name: source-provider
+    outputs:
+    - name: image
+      value: registry.example/supply-chain/my-workload@sha256:68f8e8fc6e8ede7a411db9182cd695eac7b3e7e19e4ff9dcb9ba21205c135697
+      lastTransitionTime: "2022-02-16T03:23:37Z"
+      observedGeneration: 14
+  - name: deployer
+    stampedRef:
+      apiVersion: kappctrl.k14s.io/v1alpha1
+      kind: App
+      namespace: default
+      name: my-workload
+    templateRef:
+      apiVersion: carto.run/v1
+      kind: ClusterTemplate
+      name: app-deploy
+    inputs:
+    - name: image-builder
+    observedGeneration: 19
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+- `.status.resources[*]` should match the order and length of `.spec.resource` for a ClusterSupplyChain.
+- `.status.resources[*].name` is the name of the resources as defined within the ClusterSupplyChain resource's array.
+- `.status.resources[*].templateRef` object reference to the template resource referenced by the ClusterSupplyChain.
+- `.status.resources[*].stampedRef` object reference to the Kubernetes resource created by Cartographer for this resource.
+- `.status.resources[*].inputs[*]` inputs model the relationships between resources within the SupplyChain graph. The value of an input can be read from the referenced resource's outputs.
+- `.status.resources[*].inputs[*].name` the name of the resource backing this input. 
+- `.status.resources[*].outputs[*].name` the name of the output. Output names are fixed and defined by the template type.
+- `.status.resources[*].outputs[*].value` raw value of the output. This may be large for config templates.
+- `.status.resources[*].outputs[*].lastTransitionTime` time of the most recent observed change of value. this is not the last time the value observed. This value is helpful to know if a supply chain is progressing. As this field is the result of observation of an edge triggered change, it should not be relied upon when accuracy matters. This is the same behavior as the lastTransitionTime field within a Condition.
+- `.status.resources[*].observedGeneration` the `.metadata.generation` of the stamped resource the outputs are read from. This value is not read from the `.status.observeredGeneration` field that may exist on the stamped resource. A future RFC may choose to take the stamped resource's observed generation under advisement, as that is a change to how Cartographer processes resources, it is inappropriate for this RFC.
```

Comment:
pretty interesting to have _both_ observedGeneration and lastTransitionTime - e.g., `kpack/Image` having a new build due to the builder being updated (thus, no change in generation _but_ new latestImage produced, thus update in `lastTransitionTime` despite no update in `observedGeneration`)

Reply:
The generation is always going to be spec-change centric. Outputs via the stamped resource's status can change at any time. I thought it would be interesting to capture the timestamp when the value was first observed. The timestamp is helpful to users as if the timestamp is older than you would have expected a value to be emitted, then that's a strong signal that something is wrong, even if the (future) health says it's working.